### PR TITLE
App 3 0 nested run commands

### PR DESCRIPTION
**Pull request by Kadee80**
## overview

This PR renders nested run commands as nested ordered lists in UI. Highlights current command group and subgroup. Scrolls to and highlights command being executed via isLast prop. Includes necessary style adjustments to run log and surrounding components.Was demoed at 9-20-17 sprint planning with robot and pi.

This PR also fixes progress bar layout issue seen on some previous branches including app-3-0.

This PR includes:

- [ ] Chore work
- [ ] Bug fixes
- [x] Backwards-compatible feature additions 
- [ ] Breaking changes
- [ ] Documentation updates
- [x] Test updates

## changelog

- (Feature) 
  - Render and track nested run commands in UI RunLog component. 
  - Necessary style updates. Introduce isLast to state (for tracking command currently being executed vs current command group). 
  - RunLog renders nested ordered lists. 
  - Parent-Children relationship made clear with indent classes.
- (Tests) Updated existing tests to account for isLast. 

**Note**: If your PR incorporates many changes, or many different types of changes, please consider splitting it up into smaller PRs.

## review requests
@mcous tests and lints pass, much of this was worked on in tandem, but this was rebased several times. please keep an eye out for erroneous --skips.

### CODE REVIEW

File: app/ui/components/RunLog.js [hunk 1]
```diff
@@ -9,21 +9,40 @@ export default class RunLog extends Component {
...
-      const {id, isCurrent, description} = command
-      const props = {
-        key: id,
-        className: classnames({[styles.current]: isCurrent})
+    const {style, commands} = this.props
+    const makeCommandToTemplateMapper = (depth) => (command) => {
+      const {id, isCurrent, isLast, description, children, handledAt} = command
+      const style = [styles[`indent-${depth}`]]
+      const contents = [
+        <p className={style} data-timestamp={handledAt}>[{id}] : {description}</p>
```
Comment (mcous) [inline app/ui/components/RunLog.js]: How is `data-timestamp` being used?

File: app/ui/components/RunLog.js [hunk 2]
```diff
@@ -9,21 +9,40 @@ export default class RunLog extends Component {
...
+            {children.map(makeCommandToTemplateMapper(depth + 1))}
+          </ol>
+        )
       }
-      // TODO: add ability to turn autoscroll on and off
-      if (isCurrent) props.ref = 'ensureVisible'
 
-      return (<p {...props}>[{id}] : {description}</p>)
-    })
+      const liProps = {key: id, className: classnames({[styles.executed]: handledAt, [styles.current]: isCurrent, [styles.last_current]: isLast}, style)}
```
Comment (mcous) [inline app/ui/components/RunLog.js]: Can we put a few newlines in and reorganize the classnames arguments here for readability?

``` js
const liProps = {
  key: id,
  className: classnames(style, {
    [styles.executed]: handledAt,
    [styles.current]: isCurrent,
    [styles.last_current]: isLast
  })
}
```

File: app/ui/components/RunLog.css [hunk 3]
```diff
@@ -1,15 +1,41 @@
...
-  padding: 1rem 1rem 3rem;
+  padding: 0 1rem 3rem;
   border-top: 1rem solid white;
-  border-bottom: 3rem solid white;
 }
 
-.run_log_wrapper p {
-  margin: 1rem 0;
+ol {
+  list-style-type: none;
```
Comment (mcous) [inline app/ui/components/RunLog.css]: Today I learned that `list-style` is a shorthand property for three different properties (I'd just been doing `list-style: none;` this whole time)

File: app/ui/robot/api-client/client.js [hunk 4]
```diff
@@ -5,6 +5,7 @@ import {actions, actionTypes} from '../actions'
 
 // TODO(mc, 2017-08-29): don't hardcode this URL
 const URL = 'ws://127.0.0.1:31950'
+// const URL = 'ws://238d593.local:31950'
```
Comment (mcous) [inline app/ui/robot/api-client/client.js]: We can remove this comment I think (it needs to go in state somehow)

File: app/ui/components/RunLog.js (hunk 1 above)
Comment (Kadee80) [reply app/ui/components/RunLog.js]: It was being used in CSS `content: attr(timestamp);` in an earlier version to display when a command was executed. Removed it for time being. The length of the commands and timestamp made it difficult to place. Can remove for now and possibly bring back once we format/shorten timestamps and commands

File: app/ui/components/RunLog.js (hunk 2 above)
Comment (Kadee80) [reply app/ui/components/RunLog.js]: :+1:

File: app/ui/robot/api-client/client.js (hunk 4 above)
Comment (Kadee80) [reply app/ui/robot/api-client/client.js]: :+1:

**Pull request merged**